superset-frontend/plugins/plugin-chart-echarts/test/Timeseries/transformProps.test.ts

```diff
@@ -49,8 +53,8 @@ describe('EchartsTimeseries tranformProps', () => {
   };
 
   it('should tranform chart props for viz', () => {
-    const chartProps = new ChartProps(chartPropsConfig);
-    expect(transformProps(chartProps)).toEqual(
+    const chartProps: unknown = new ChartProps(chartPropsConfig);
```

Comment:
There are [a few prop types described in this component](https://github.com/apache/superset/blob/6f8927209b80dc61ba14fc66ff94871b36a83321/superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts#L32-L37), would any of those work here instead of `unknown`?

Reply:
I am actually not sure what is going on with these. I set it to unknown because TS said I had to before it would let me do the type assertion here `chartProps as EchartsTimeseriesChartProps`. It throws an error when I try to set the type of chartProps to EchartsTimeseriesChartProps when it is defined because their formData prop types are incompatible. This is all strange because they don't actually appear to be incompatible because EchartsTimeseriesChartProps is an extension of ChartProps and ChartProps formData prop is set to be an ambiguous object.

This all appears to be done intentionally and I am not sure what I should do to get rid of the errors because I haven't found a similar test file where this isn't causing TS errors. Give me a little time to look into this more because I am not sure what to do at the moment

Reply:
This specific line should be something like
```typescript
    const chartProps = new ChartProps<EchartsTimeseriesChartProps>(chartPropsConfig);
```
However, it seems these types are slightly tangled up right now, and should probably be addressed in a follow-up PR, as I believe it's going to require some more type fixing. I'd also want to enable proper type checking for the test folders, but that also requires a separate PR.